### PR TITLE
Use Assert::lazy for exception throwing #37

### LINKED ISSUE #37: Update exception handling for static saving method
Refer to https://github.com/tipoff/addresses/pull/57#discussion_r591677500 for refactoring how exceptions are handled in package models

### PR DESCRIPTION
**Pull request by chx2**
Closes #37

### CODE REVIEW

File: src/Models/Refund.php [hunk 1]
```diff
@@ -38,12 +39,11 @@ public static function boot()
...
-            }
-            if (empty($refund->amount)) {
-                throw new \Exception('A refund must be for an amount.');
-            }
+            Assert::lazy()
+                ->that($refund->payment_id)->notEmpty('A refund must be applied to a payment.')
+                ->that($refund->amount)->notEmpty('A refund must be for an amount.')
+                ->verifyNow();
+            //Todo: Add the following to the assert if possible
             if ($refund->amount > ($refund->payment->amount_refundable)) {
```
Comment (chx2) [inline src/Models/Refund.php]: @pdbreen is there an assert method that I could use to to evaluate a conditional statement like the one shown?

`$refund->amount > ($refund->payment->amount_refundable)`

Comment (pdbreen) [reply src/Models/Refund.php]: `->that($refund->amount > ($refund->payment->amount_refundable))->false('message')`

Comment (pdbreen) [reply src/Models/Refund.php]: Might be better to use `lessOrEqualThan`  -- I think like this, but not at my desk so can't confirm exact syntax

 `->that($refund->amount)->lessOrEqualThan($refund->payment->amount_refundable, 'message')`

https://github.com/beberlei/assert#list-of-assertions

Comment (chx2) [reply src/Models/Refund.php]: Thank you!

**Review by drewroberts — APPROVED** [review]
Thank you! 🚚

**Pull request merged**